### ISSUE TITLE
Why my mAP is not increasing?

### ISSUE THREAD

**Issue by PumayHui** [AlexeyAB/darknet#2150]
Hi, @AlexeyAB ,
Such as the picture, I ensure the data is correct, but the test mAP is so low and not increasing. Could you help me? Thanks so much!
![default](https://user-images.githubusercontent.com/28421690/50736525-8e185900-11f9-11e9-9fe6-3e558b927fa4.png)

And the cfg file is : 
![default](https://user-images.githubusercontent.com/28421690/50736602-31696e00-11fa-11e9-98fa-d80d0847c485.png)

**Comment by AlexeyAB** [comment]
@PumayHui Hi,

> I ensure the data is correct

* Did you check your Training and Valid datasets by using Yolo_mark? Also just check `bad.list` and `bad_label.list`

* Does your training dataset contain valid dataset?

* How many images in your Training and Valid datasets?

* How many classes in your cfg-file?

* What GPU do you use? And what params in Makefile do you use?

* Did you recalculate anchors?

* Can you show **screenshot** of output of the command?
`./darknet detector calc_anchors data/obj.data -num_of_clusters 9 -width 512 -height 512 -show`

* Then set `train=valid.txt` in your `obj.data` and show **screenshot** of output of the command
`./darknet detector calc_anchors data/obj.data -num_of_clusters 9 -width 512 -height 512 -show`

**Comment by PumayHui** [comment]
- I have not check by Yolo_mark. I just try to check this.
- No.
- Training dataset : 280,000 pictures
   Valid dataset: 1000 pictures
- 5 classes.
- GTX 1080Ti
  GPU=1
  CUDNN=1
  OPENCV=1
  OPENMP=0
  DEBUG=0
- No.
- 
![default](https://user-images.githubusercontent.com/28421690/50737047-ce7ad580-11ff-11e9-879e-c44e8186f706.png)
-
![default](https://user-images.githubusercontent.com/28421690/50737064-04b85500-1200-11e9-85ce-59986237011f.png)

Thanks so much!!!

**Comment by AlexeyAB** [comment]
> Training dataset : 280,000 pictures
> Valid dataset: 1000 pictures

It is recommended to use Valid/Train = from 5% / 95% to 20%/80%
So try to increase number of images in Validation dataset up to ~10 000 images.
1000 images may be not enough to calculate correct mAP.

----

Do you have `bad.list` or `bad_label.list` files?

----

> I have not check by Yolo_mark. I just try to check this.

Yes, try to check it.
It seems that many objects in your dataset have zero size (or smaller than width=1 height=1).

> - 
> ![default](https://user-images.githubusercontent.com/28421690/50737047-ce7ad580-11ff-11e9-879e-c44e8186f706.png)
> -
> ![default](https://user-images.githubusercontent.com/28421690/50737064-04b85500-1200-11e9-85ce-59986237011f.png)
> 

----

Did you set `CUDNN_HALF=0` in the Makefile?


----

Try to run these commands with `-show` flag and show me **2 screenshots of windows with cloud of points for Train and Valid datasets**:
`./darknet detector calc_anchors data/obj.data -num_of_clusters 9 -width 512 -height 512 -show`

**Comment by PumayHui** [comment]
Hi, I run the same calc_anchors command in the version 'darknet-AB-2018-08-22', and the results are:
![default](https://user-images.githubusercontent.com/28421690/50737617-a3e04b00-1206-11e9-8d99-347cd573b826.png)

**Comment by PumayHui** [comment]
And the bad.list and bad_label.list are None.

**Comment by PumayHui** [comment]
Did you set CUDNN_HALF=0 in the Makefile?
- No.
  Does this has any influence?

**Comment by eonia89** [comment]
@PumayHui Yes, sir
http://on-demand.gputechconf.com/gtc/2017/presentation/s7218-training-with-mixed-precision-boris-ginsburg.pdf

**Comment by PumayHui** [comment]
Actually, the Makefile is such as:
GPU=1
CUDNN=1
CUDNN_HALF=0
OPENCV=1
AVX=0
OPENMP=0
LIBSO=0

Is there any problems?
Thanks so much!
@AlexeyAB @eonia89

**Comment by AlexeyAB** [comment]
@PumayHui 
This is correct.

Just try to use these recommendations to increase mAP: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

**Comment by leo-XUKANG** [comment]
the loss and MAP are not strictly linear relationships,

**Comment by AlexeyAB** [comment]
And use **more than 10 000 images for validation dataset.**

**Comment by PumayHui** [comment]
Well, thanks so much!
